internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -191,3 +181,52 @@ func (c *CodePipeline) getPipelineName(resourceArn string) (string, error) {
 
 	return parsedArn.Resource, nil
 }
+
+// GetPipelineStatus retrieves status information from a given pipeline.
+func (c *CodePipeline) GetPipelineState(name string) (*PipelineState, error) {
+	input := &cp.GetPipelineStateInput{
+		Name: aws.String(name),
+	}
+	resp, err := c.client.GetPipelineState(input)
+
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline state %s: %w", name, err)
+	}
+	var stageStates []*StageState
+	for _, stage := range resp.StageStates {
+		var bool string
+		if *stage.InboundTransitionState.Enabled == true {
+			bool = "ENABLED"
+		} else {
+			bool = "DISABLED"
+		}
+		var status string
+		for _, actionState := range stage.ActionStates {
+			if actionState.LatestExecution == nil {
+				status = "  -"
```

Comment:
Maybe `const emptyStatus = "  -"` then refer the const here for better readability?

Reply:
I'll also use it for Transition if InboundTransitionState is not present.

Reply:
Q: should HumanString handle the "   -" instead, because as is, "  -" is appearing in JSON output (and will never omitempty)?